src/simulation/realized_meta.jl

```diff
@@ -69,6 +69,8 @@ function _make_dataframe(
         row_end = row_index + last_id - first_id
         matrix[row_index:row_end, :] = array[first_id:last_id, :]
         row_index += last_id - first_id + 1
+        # Gracefully handle being given more results_by_time than we need
```

Comment:
When does this happen and why isn’t it an error?

Reply:
This is an interface issue @daniel-thom discussed where we pass a store to read from or nothing when nothing signifies that it is an in-memory store which makes the code confusing since it isn't obvious.

Reply:
`_read_results` should only respond with the requested data. That’s a bug.

Reply:
Okay, will fix

Reply:
https://github.com/NREL-Sienna/PowerSimulations.jl/issues/1074